src/network/networkd-link.c

```diff
@@ -1357,16 +1357,18 @@ static int link_set_flags(Link *link) {
         if (!link->network)
                 return 0;
 
-        if (link->network->arp < 0)
-                return 0;
-
         r = sd_rtnl_message_new_link(link->manager->rtnl, &req, RTM_SETLINK, link->ifindex);
         if (r < 0)
                 return log_link_error_errno(link, r, "Could not allocate RTM_SETLINK message: %m");
 
         if (link->network->arp >= 0) {
                 ifi_change |= IFF_NOARP;
-                ifi_flags |= link->network->arp ? 0 : IFF_NOARP;
+                ifi_flags |= link->network->arp ? ~IFF_NOARP : IFF_NOARP;
```

Comment:
I guess this should be the following:
```
if (link->network->arp)
   ifi_flags &= ~IFF_NOARP;
else
   ifi_flags |= IFF_NOARP;
```
Or, more simply, use `SET_FLAG()` macro.